### DISCUSSION TITLE
how to remove focus from a q-tabs group

### DISCUSSION THREAD

**Discussion by telethonic** [quasarframework/quasar#16169] [General - Components / Directives / etc]
Hi,
I use a tabs group to select a difficulty level:
```
<q-tabs v-model="selected_tab" @update:model-value="change_level">
    <q-tab name="easy" icon="star_outline" label="Easy" />
    <q-tab name="medium" icon="balance" label="Medium" />
    <q-tab name="hard" icon="school" label="Hard" />
</q-tabs>
```

For a puzzle grid below it I have a keydown handler set up:
```
window.addEventListener('keydown', (event) => {
    process_key_down(event);
  });
```

The problem now is that after the user selects a tab, say "medium", the tab group has focus:
the left and right arrow key now get handled by the tab group, and no longer move the cursor in the puzzle.

Is there a way to remove focus from the tab group?

I'm using the composition API with `<script setup>`

Thanks :)

**Comment by telethonic** [comment]
From the code in Qtabs.js:
```
 function onKbdNavigate (keyCode, fromEl) {
      const tabs = Array.prototype.filter.call(
        contentRef.value.children,
        el => el === fromEl || (el.matches && el.matches('.q-tab.q-focusable') === true)
      )

      const len = tabs.length
      if (len === 0) { return }
```

It looks like if I remove the `q-focusable` class from each tab, then keyboard navigation will be disabled
(added `id="easy_tab"` for example, then:
 `document.getElementById('easy_tab').classList.remove('q-focusable');`).

The only issue now is that after a tab is selected (by a mouse click, for example), the "old tab" and the selected tab get the `q-focusable` class again (!)

After trying some stuff, a solution that seems to work: 

- add a `tabindex` attribute to the div surrounding the puzzle
- call `focus()` on it everytime the tab changes: `document.getElementById('puzzle_div').focus();`

**Comment by JeanMg26** [comment]
only add props in tab :ripple="false"

**Reply by telethonic** [reply]
But 'ripple' is a single tab style property?

How would that affect focus?
